### ISSUE TITLE
seq_linter() gives the wrong message for n:1 lints

### ISSUE THREAD

**Issue by MichaelChirico** [r-lib/lintr#1542]
The result message should mention `rev()`, but currently it doesn't:

```
lint(text = ".N:1", linters = seq_linter())
# <text>:1:1: warning: [seq_linter] .N:1 is likely to be wrong in the empty edge case. Use seq_len(1) instead.
# .N:1
# ^~~~
```

It also says `seq_len(1)` instead of `seq_len(.N)`

**Comment by IndrajeetPatil** [comment]
The same issue also holds for all other relevant functions I think:

``` r
library(lintr)

lint(text = "length(x):1", linters = seq_linter())
#> <text>:1:1: warning: [seq_linter] length(...):1 is likely to be wrong in the empty edge case. Use seq_along(...) instead.
#> length(x):1
#> ^~~~~~~~~~~

length(BOD):1
#> [1] 2 1

rev(seq_along(BOD))
#> [1] 2 1
```

<sup>Created on 2022-09-20 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

We have a TODO to track this:

https://github.com/r-lib/lintr/blob/a8d5d1f68b824fc335f69654bc62a7646bdad8ef/R/seq_linter.R#L67-L68

**Comment by MichaelChirico** [comment]
Also for viz: trying to get r-core to give me more confidence in recommending a replacement in the `rev()` case:

https://bugs.r-project.org/show_bug.cgi?id=18406
